src/lib/y2packager/known_repositories.rb

```diff
@@ -70,10 +70,15 @@ def read_repositories
         return []
       end
 
-      status = YAML.load_file(status_file)
+      begin
+        status = YAML.load_file(status_file)
+      rescue Psych::SyntaxError
+        log.error("Status file #{status_file} is corrupted")
```

Comment:
Looks good, just please also log exception content, so we know how it is corrupted.